src/targets/cpu/gemm.cpp

```diff
@@ -43,7 +43,10 @@ struct dnnl_gemm : dnnl_extend_op<dnnl_gemm, dnnl::matmul, op::dot>
                 MIGRAPHX_DNNL_PREFIX(ARG_BIAS)};
     }
 
-    void required(const check_shapes& cs) const { cs.not_broadcasted(); }
+    void required(const check_shapes<std::vector<shape>::const_iterator>& cs) const
```

Comment:
It would be better to template this:

```cpp
template<class T>
void required(const check_shapes<T>& cs) const
```

Reply:
Would have to make a header file to do that, are we OK with that?

Reply:
No need to make a header, this is not used directly outside of the cpp file.